crates/ilp-node/benches/multiple_payments.rs

```diff
@@ -0,0 +1,386 @@
+            node_b.serve(None).await.unwrap();
+            create_account_on_node(node_b_http, a_on_b, "admin")
+                .await
+                .unwrap();
+            create_account_on_node(node_b_http, bob_on_b, "admin")
+                .await
+                .unwrap();
+
+            // start node a and open its accounts
+            node_a.serve(None).await.unwrap();
+            create_account_on_node(node_a_http, alice_on_a, "admin")
+                .await
+                .unwrap();
+            create_account_on_node(node_a_http, b_on_a, "admin")
+                .await
+                .unwrap();
+        },
+    );
+
+    let ws_request = Request::builder()
+        .uri(format!("ws://localhost:{}/payments/incoming", node_b_http))
+        .header("Authorization", "Bearer admin")
+        .body(())
+        .unwrap();
+    let (sender, mut receiver) = channel(RECEIVED_MESSAGE_COUNT_HIGH);
+    let client = reqwest::Client::new();
+    let req_low = client
+        .post(&format!(
+            "http://localhost:{}/accounts/{}/payments",
+            node_a_http, "alice_on_a"
+        ))
+        .header("Authorization", format!("Bearer {}", "admin"))
+        .json(&json!({
+            "receiver": format!("http://localhost:{}/accounts/{}/spsp", node_b_http,"bob_on_b"),
+            "source_amount": 100,
+            "slippage": 0.025 // allow up to 2.5% slippage
+        }));
+    let req_high = client
+        .post(&format!(
+            "http://localhost:{}/accounts/{}/payments",
+            node_a_http, "alice_on_a"
+        ))
+        .header("Authorization", format!("Bearer {}", "admin"))
+        .json(&json!({
+            "receiver": format!("http://localhost:{}/accounts/{}/spsp", node_b_http,"bob_on_b"),
+            "source_amount": 10000,
+            "slippage": 0.025 // allow up to 2.5% slippage
+        }));
+    let handle = std::thread::spawn(move || payment_notification_forwarder(ws_request, sender));
+    wait_for_propagation(&mut rt, &req_low, RECEIVED_MESSAGE_COUNT_LOW, &mut receiver);
+    c.bench_function("process_payment_http_single_packet", |b| {
+        b.iter(|| bench_fn(&mut rt, &req_low, RECEIVED_MESSAGE_COUNT_LOW, &mut receiver));
+    });
+    c.bench_function("process_payment_http_hundred_packets", |b| {
+        b.iter(|| {
+            bench_fn(
+                &mut rt,
+                &req_high,
+                RECEIVED_MESSAGE_COUNT_HIGH,
+                &mut receiver,
+            )
+        });
+    });
+
+    drop(rt);
+    handle.join().unwrap().unwrap();
+}
+
+/// Forwards the payment notifications from the websocket into the channel
+/// When runtime (rt) is dropped, error will happen ending the loop and then the websocket
+fn payment_notification_forwarder(
+    ws_request: tungstenite::http::Request<()>,
+    mut sender: tokio::sync::mpsc::Sender<tungstenite::Message>,
+) -> Result<(), tungstenite::Error> {
+    let mut payments_ws = client::connect(ws_request).unwrap().0;
```

Comment:
```suggestion
    let mut payments_ws = client::connect(ws_request)?.0;
```

I think this is a leftover `unwrap`.